crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -149,7 +149,11 @@ def ceph_status
         return ret if ceph_nodes.empty?
         ceph_node = ceph_nodes.first
 
+        # FIXME: do I need to pick ceph-mon node? (It might happen that this is just ceph-osd, without the mon role)
         ssh_retval = ceph_node.run_ssh_cmd("LANG=C ceph health --connect-timeout 5 2>&1")
+        # FIXME: here we need some better check for health:
+        # a) could we ignore HEALTH_WARN in general, thus only checking for HEALTH_ERR?
+        # b) do we need some magic to suppress only some kind of warnings, making the cluster return HEALTH_OK?
```

Comment:
What @smithfarm suggested can be implemented something like this:

````
#!/usr/bin/env ruby

require 'json'

# `ceph daemon` doesn't need --connect-timeout as it's
# talking directly to the daemon on this node only to
# get a config setting.  If the mon isn't running, this'll
# return something ugly and the JSON parse will throw an
# exception.
warn_max = JSON.parse(`ceph daemon mon.$(hostname) config get mon_pg_warn_max_per_osd`)['mon_pg_warn_max_per_osd']
warn_min = JSON.parse(`ceph daemon mon.$(hostname) config get mon_pg_warn_min_per_osd`)['mon_pg_warn_min_per_osd']

# have to tell *all* the mons the new config value, not
# just the current mon (experimentation indicates using
# `ceph daemon mon.$(hostname) config set ...` doesn't
# always result in `ceph health` noticing the value is
# set yet
`ceph tell 'mon.*' injectargs '--mon_pg_warn_max_per_osd 0' >/dev/null 2>&1`
`ceph tell 'mon.*' injectargs '--mon_pg_warn_min_per_osd 0' >/dev/null 2>&1`

rv = `LANG=C ceph health --connect-timeout 5 2>&1`

# back to previous values
`ceph tell 'mon.*' injectargs '--mon_pg_warn_max_per_osd #{warn_max}' >/dev/null 2>&1`
`ceph tell 'mon.*' injectargs '--mon_pg_warn_min_per_osd #{warn_min}' >/dev/null 2>&1`

if rv.include? "HEALTH_OK"
  print "cluster healthy\n"
else
  print "cluster not healthy\n"
end
````

That looks kinda complicated to me though. I'm going to ask about this upstream and see if anyone knows a better way.